src/foam/u2/view/EditColumnsView.js

```diff
@@ -4,84 +4,118 @@
+      class: 'FObjectArray',
+      of: 'foam.u2.view.ColumnConfig',
+      name: 'columns',
+      view: {
+        class: 'foam.u2.view.FObjectArrayView',
+        valueView: {
+          class: 'foam.u2.detail.SectionView',
+          sectionName: '_defaultSection'
+        },
+        mode: 'RO'
+      },
+      factory: function() {
+        return this.allColumns
+          .map(c => {
+            var a = this.of.getAxiomByName(c);
+            var v = localStorage.getItem(this.of.id + '.' + a.name);
+            v = this.ColumnVisibility.VALUES.find(e => e.name == v);
+            return this.ColumnConfig.create({
+              name: a.name,
+              label: a.label || foam.String.labelize(a.name),
+              visibility: v || 'DEFAULT'
+            })
+          })
+      }
     },
     {
-      name: 'columns_'
+      class: 'StringArray',
+      name: 'allColumns',
+      factory: function() {
+        var props = this.of.getAxiomsByClass(foam.core.Property)
+          .filter(p => p.tableCellFormatter && ! p.hidden)
+          .map(p => p.name);
+        var actions = this.of.getAxiomsByClass(foam.core.Action)
+          .map(a => a.name);
+        return props.concat(actions);
+      },
+      hidden: true
     }
   ],
-
-  css: `
-    ^ {
-      text-align: left;
-      padding: 10px;
-      width: 125px;
-    }
-    ^ label {
-      margin-top: 5px;
-    }
-  `,
-
-  methods: [
-    function initE() {
-      var index = 0;
-      this
-        .addClass(this.myClass())
-        .forEach(this.columns, function(columnName) {
-          var axiom = typeof columnName === 'string'
-            ? this.table.getAxiomByName(columnName)
-            : columnName;
-          var localStorageKey = this.table.id + '.' + columnName;
-
-          var slot = this.SimpleSlot.create();
-
-          this
-            .start()
-              .tag(this.CheckBox, {
-                label: axiom.label,
-                data: localStorage.getItem(localStorageKey) === null
-              }, slot)
-            .end();
-
-          this.onDetach(slot.value.data$.sub(this.updateCols(axiom, index++, localStorageKey)));
-        });
-    }
+  // This shouldn't be needed.
+  imports: [
+    'stack'
   ],
-
-  listeners: [
-    function updateCols(axiom, index, localStorageKey) {
-      return (_, __, ___, propSlot) => {
-        var checked = propSlot.get();
-        if ( checked ) {
-          localStorage.removeItem(localStorageKey);
-
-          // Put the column back in the correct position.
-          this.columns_.splice(index, 0, axiom);
-
-          // Force the view to update.
-          this.columns_ = this.columns_.slice();
-        } else {
-          // The string isn't important, all that matters is that there's some
-          // value for this key.
-          localStorage.setItem(localStorageKey, 'Y');
-          this.columns_ = this.columns_.filter((col) => col.name !== axiom.name);
-        }
-      };
+  actions: [
```

Comment:
An action to set all to default might be useful. Also a cancel button (which shouldn't be needed once we give the stackview a back button)